xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/cssVariablesInit.css

```diff
@@ -0,0 +1,204 @@
+#foreach($excludedVariable in $excludeFromOldColorThemeMapping)
+  #set($discard = $oldColorThemeVariables.remove($excludedVariable))
+#end
+
+## We merge the three maps into one list.
+## We allow different values from different maps for the same variable. This is an extra fallback in case the first
+## doesn't work. This could be the case for defaults that rely on other variables.
+## Values follow CSS cascading order (last one wins).
+#set($variableValueSets = [$oldColorThemeVariables, $bootstrapVariables, $xsVariables])
+
+## $flamingoColorThemeVariables is the list of all the variables available to customize directly in the Flamingo
+## theme editor. These mappings will not always end up in the initialization of a CSS variable.
+## If you want to update the values set for the flamingo colortheme, see the colortheme xml,
+## a source of truth shared with LESS variables, defined in xwiki-platform-flamingo-theme-ui
+## To make sure all of those variables are still available even when not set in the colortheme,
+## we make sure that they all appear in another mapping above.
+## The key is the name and the value is a type for the variable.
+######################
+## Generation
+######################
+
+#define($css)
+## We use a tan/arctan2 hack to get a proper unitless viewport width. This hack can be avoided when the CSS Values and
+## Units module Level 4 is accepted and supported in browsers.
+## https://drafts.csswg.org/css-values/#calc-type-checking
+@property --100vw {
+  syntax: "<length>";
+  initial-value: 0px;
+  inherits: false;
+}
+
+:root {
+## First, we make sure that the necessary variables from bootstrap and XS are given at least one value.
+#foreach($variableWithValue in $variableValueSets)
+  #foreach($variableName in $variableWithValue.keySet())
+    #set($value = $variableWithValue.get($variableName))
+    #if("$!value" != '')
+      --$variableName: $!value;
+    #end
+  #end
+#end
+
+## Create CSS variables for each flamingo colorTheme variable that is set.
+## Right now we have the variable type info for all of those, but we don't use those yet.
+## Those values will override anything already declared in the XS/bootstrap defaults.
+#foreach($property in $flamingoColorThemeVariables.keySet())
+#set($value = $themeDoc.getValue($property, $themeObj))
+#if("$!value" != '')
+## If the value we want to set is a LESS variables, we make sure to convert it into a CSS variable.
+#if($stringtool.startsWith($!value, '@'))
+#set($value = 'var(--' + $stringtool.substring($value, 1) + ')')
+#end
+#set($propertyType = $flamingoColorThemeVariables.get($property))
+## For now we only use `string` variables, because it's the easiest to setup and because it's 100% compatible with
+## LESS variables. We could use the extra info we have on the variable types to use the @property syntax later.
+#if($propertyType == 'escapedText')
+--$property: "$escapetool.javascript($!value)";
+#else
+--$property: $!value;
+#end
+#end
+#end
+}
+#end
+*/$!css
```

Comment:
This works, minification doesn't remove the whole comment?

Reply:
The comment with velocity pattern is inspired from https://github.com/xwiki/xwiki-platform/blob/226762d964f8a4ba298d84a3bacad3eb42efed03/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/style.css#L10
afaik the content of the comment is kept during minification.
Apparently, the magic in the exclamation mark at the end of the opening on the comment (`/*!`)
I tried locally, and removing the exclamation mark makes the comment disappear during minification.
Maybe @mflorea knows where this comes from?

Reply:
> Maybe @mflorea knows where this comes from?

@manuelleduc this is most probably needed to avoid YuiCompressor from removing the comment (with the Velocity code) when minifying the CSS file. The alternative is to exclude this CSS file from minification (in the POM).